lib/kubernetes-deploy/deploy_task.rb

```diff
@@ -204,7 +203,18 @@ def run!(verify_result: true, allow_protected_ns: false, prune: true)
     private
 
     def cluster_resource_discoverer
-      ResourceDiscovery.new(namespace: @namespace, context: @context, logger: @logger, namespace_tags: @namespace_tags)
+      @cluster_resource_discoverer ||= ResourceDiscovery.new(
+        namespace: @namespace,
+        context: @context,
+        logger: @logger,
+        namespace_tags: @namespace_tags
+      )
+    end
+
+    def add_custom_resources_to_predeploy_sequence(resources)
+      crds = cluster_resource_discoverer.crds(@sync_mediator).map(&:kind)
+      discovered_crs = resources.select { |resource| crds.include? resource.type }
```

Comment:
nit: looks like you could do the `.map(&type).uniq` before the `select` as well to reduce the potentially large set of resources into just their unique types